providers/ms365/resources/policies.go

```diff
@@ -110,3 +113,73 @@ func (a *mqlMicrosoftPolicies) consentPolicySettings() (interface{}, error) {
 
 	return convert.JsonToDict(actualSettingsMap)
 }
+
+func (a *mqlMicrosoftPolicies) authenticationMethodsPolicy() (*mqlMicrosoftPoliciesAuthenticationMethodsPolicy, error) {
+	conn := a.MqlRuntime.Connection.(*connection.Ms365Connection)
+	graphClient, err := conn.GraphClient()
+	if err != nil {
+		return nil, err
+	}
+
+	ctx := context.Background()
+	// expand authenticationMethodConfigurations to get all the details in one call
+	requestConfiguration := &policies.AuthenticationMethodsPolicyRequestBuilderGetRequestConfiguration{
+		QueryParameters: &policies.AuthenticationMethodsPolicyRequestBuilderGetQueryParameters{
+			Expand: []string{"authenticationMethodConfigurations"},
+		},
+	}
+
+	resp, err := graphClient.Policies().AuthenticationMethodsPolicy().Get(ctx, requestConfiguration)
+	if err != nil {
+		return nil, transformError(err)
+	}
+
+	return a.newAuthenticationMethodsPolicy(resp)
+}
+
+func (a *mqlMicrosoftPolicies) newAuthenticationMethodsPolicy(policy models.AuthenticationMethodsPolicyable) (*mqlMicrosoftPoliciesAuthenticationMethodsPolicy, error) {
+	authMethodConfigs, err := a.newAuthenticationMethodConfigurations(policy.GetAuthenticationMethodConfigurations())
+	if err != nil {
+		return nil, err
+	}
+
+	mqlAuthenticationMethodsPolicy, err := CreateResource(a.MqlRuntime, "microsoft.policies.authenticationMethodsPolicy",
+		map[string]*llx.RawData{
+			"description":                        llx.StringDataPtr(policy.GetDescription()),
+			"displayName":                        llx.StringDataPtr(policy.GetDisplayName()),
+			"id":                                 llx.StringDataPtr(policy.GetId()),
+			"lastModifiedDateTime":               llx.TimeDataPtr(policy.GetLastModifiedDateTime()),
+			"policyVersion":                      llx.StringDataPtr(policy.GetPolicyVersion()),
+			"authenticationMethodConfigurations": llx.ArrayData(authMethodConfigs, "microsoft.policies.authenticationMethodConfiguration"),
+		})
+	if err != nil {
+		return nil, err
+	}
+
+	return mqlAuthenticationMethodsPolicy.(*mqlMicrosoftPoliciesAuthenticationMethodsPolicy), nil
+}
+
+func (a *mqlMicrosoftPolicies) newAuthenticationMethodConfigurations(configs []models.AuthenticationMethodConfigurationable) ([]interface{}, error) {
```

Comment:
I would not define this as an extenstion. You only use the MqlRuntime here from the mqlMicrosoftPolicies struct. Why not just make the runtime a parameter for the function?